File: intrabank-frontend/src/app/core/service/document/category.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { environment } from "../../../../environments/environment.development";
import { CategoryRequest, CategoryResponse } from "../../../shared/models/category.model";

@Injectable({
  providedIn: "root",
})
export class CategoryService {
  constructor(private http: HttpClient) {
  }

  // Get all categories
  getAllCategories(): Observable<CategoryResponse[]> {
    const url = `${environment.apiUrl}/users/categories`;
    return this.http.get<CategoryResponse[]>(url);
  }

  // Get category by ID
  getCategoryById(id: string): Observable<CategoryResponse> {
    const url = `${environment.apiUrl}/users/categories/${id}`;
    return this.http.get<CategoryResponse>(url);
  }

  // Create a new category
  createCategory(category: CategoryRequest): Observable<CategoryResponse> {
    const url = `${environment.apiUrl}/admin/categories`;
    return this.http.post<CategoryResponse>(url, category);
  }

  // Update category
  updateCategory(category: CategoryRequest): Observable<CategoryResponse> {
    const url = `${environment.apiUrl}/admin/categories/${category.id}`;
    return this.http.put<CategoryResponse>(url, category);
  }

  // Delete category
  deleteCategory(id: string): Observable<void> {
    const url = `${environment.apiUrl}/admin/categories/${id}`;
    return this.http.delete<void>(url);
  }
} 

File: intrabank-backend/src/main/java/com/intrabank/intrabank_backend/service/CategoryService.java
package com.intrabank.intrabank_backend.service;

import com.intrabank.intrabank_backend.dto.request.CategoryRequest;
import com.intrabank.intrabank_backend.dto.response.CategoryResponse;
import com.intrabank.intrabank_backend.model.Category;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface CategoryService {

    Optional<Category> findById(UUID id);
    
    CategoryResponse createCategory(CategoryRequest request);
    
    CategoryResponse getCategoryById(UUID id);
    
    List<CategoryResponse> getAllCategories();
    
    CategoryResponse updateCategory(UUID id, CategoryRequest request);
    
    void deleteCategory(UUID id);
} 